spec.md

```diff
@@ -286,6 +286,67 @@ circuit Foo :
   public module Foo enablelayer A :
 ```
 
+## Formal Tests
+
+Formal tests are named top-level constructs which allow for formal test harnesses to be definied 
+as bounded model checking (BMC) problems. The body of the formal test harness is converted 
+into a BMC formula which encodes a state-transition system version of the design using one 
+of the formal backends (targetting either BTOR2 or SMTLib). 
+Formal tests have 3 operands: 
+- A symbol that the test can be referred to with.  
+- The name of the formal test harness module.  
+- A bound for the bounded model checking problem, which refers to the maximum bound k used
+to check the bounded model checking formula. This means that given a bound k, the BMC formula 
+will be checked for all unrollings of the resulting state-transition system of depths 1 through k.
+This bound is a strictly positive integer.  
```

Comment:
This probably needs to be further layicized, since it introduces a number of concepts beyond the scope of the FIRRTL spec.

Could I suggest something like:

> "A positive integer bound for test. This number indicates the depth to which the model will be checked."